src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs

```diff
@@ -0,0 +1,204 @@
+            sourceLine = 0;
+            sourceColumn = 0;
+
+            MetadataReader reader;
+            using (GetReader(assemblyPath, inMemoryAddress, inMemorySize, out reader))
+            {
+                if (reader != null)
+                {
+                    Handle handle = MetadataTokens.Handle(methodToken);
+
+                    if (handle.Kind == HandleKind.MethodDefinition)
+                    {
+                        MethodDebugInformationHandle methodDebugHandle = ((MethodDefinitionHandle)handle).ToDebugInformationHandle();
+                        MethodDebugInformation methodInfo = reader.GetMethodDebugInformation(methodDebugHandle);
+
+                        if (!methodInfo.SequencePointsBlob.IsNil)
+                        {
+                            SequencePointCollection sequencePoints = methodInfo.GetSequencePoints();
+
+                            SequencePoint? bestPointSoFar = null;
+                            foreach (SequencePoint point in sequencePoints)
+                            {
+                                if (point.Offset > ilOffset)
+                                    break;
+
+                                if (point.StartLine != SequencePoint.HiddenLine)
+                                    bestPointSoFar = point;
+                            }
+
+                            if (bestPointSoFar.HasValue)
+                            {
+                                sourceLine = bestPointSoFar.Value.StartLine;
+                                sourceColumn = bestPointSoFar.Value.StartColumn;
+                                sourceFile = reader.GetString(reader.GetDocument(bestPointSoFar.Value.Document).Name);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the portable PDB reader for the assembly path
+        /// </summary>
+        /// <param name="assemblyPath">file path of the assembly or null</param>
+        /// <param name="inMemoryAddress">in memory PE (assemblyPath != null) or PDB (assemblyPath == null) address or zero</param>
+        /// <param name="inMemorySize">in memory PE or PDB size</param>
+        /// <param name="reader">returns the reader</param>
+        /// <returns>provider</returns>
+        static IDisposable GetReader(string assemblyPath, IntPtr inMemoryAddress, int inMemorySize, out MetadataReader reader)
+        {
+            reader = null;
+            MetadataReaderProvider provider = null;
+
+            if (assemblyPath != null)
+            {
+                uint stamp;
+                int age;
+                Guid guid;
+
+                string pdbName = GetPdbPathFromPeStream(assemblyPath, inMemoryAddress, inMemorySize, out age, out guid, out stamp);
+                if (pdbName != null)
+                {
+                    try
+                    {
+                        Stream pdbStream = File.OpenRead(pdbName);
```

Comment:
Oh, nevermind, I see the comment below and that we're always returning the provider that wraps the stream.